examples/step-82/step-82.cc

```diff
@@ -0,0 +1,1500 @@
+
+                    for (unsigned int m = 0; m < n_dofs_lift; ++m)
+                      {
+                        local_rhs_re(m) +=
+                          0.5 * (fe_face_lift[tau_ext].value(m, q) * normal) *
+                          fe_face_neighbor.shape_grad(i, q) * dx;
+                      }
+                  }
+
+                // Here, <code>local_rhs_be(m)</code> corresponds to $G_m$
+                // introduced in the comments about the implementation of the
+                // lifting $b_e$ in the case
+                // $\varphi=\varphi^n$.
+                local_rhs_be = 0;
+                for (unsigned int q = 0; q < n_q_points_face; ++q)
+                  {
+                    const double         dx = fe_face_lift.JxW(q);
+                    const Tensor<1, dim> normal =
+                      fe_face_neighbor.normal_vector(q);
+
+                    for (unsigned int m = 0; m < n_dofs_lift; ++m)
+                      {
+                        local_rhs_be(m) +=
+                          0.5 * fe_face_lift[tau_ext].divergence(m, q) *
+                          normal * fe_face_neighbor.shape_value(i, q) * dx;
+                      }
+                  }
+
+                solver.solve(local_matrix_lift,
+                             coeffs_re,
+                             local_rhs_re,
+                             PreconditionIdentity());
+                solver.solve(local_matrix_lift,
+                             coeffs_be,
+                             local_rhs_be,
+                             PreconditionIdentity());
+
+                for (unsigned int q = 0; q < n_q_points; ++q)
+                  {
+                    for (unsigned int m = 0; m < n_dofs_lift; ++m)
+                      {
+                        discrete_hessians_neigh[face_no][i][q] -=
+                          coeffs_re[m] * fe_values_lift[tau_ext].value(m, q);
+                      }
+
+                    for (unsigned int m = 0; m < n_dofs_lift; ++m)
+                      {
+                        discrete_hessians_neigh[face_no][i][q] +=
+                          coeffs_be[m] * fe_values_lift[tau_ext].value(m, q);
+                      }
+                  }
+
+              } // for dof i
+          }     // boundary check
+      }         // for face
+  }
+
+
+
+  // @sect4{BiLaplacianLDGLift::run}
+  template <int dim>
+  void BiLaplacianLDGLift<dim>::run()
+  {
+    make_grid();
+
+    setup_system();
+    assemble_system();
+
+    solve();
+
+    compute_errors();
+    output_results();
+  }
+
+} // namespace Step82
+
+
+
+// @sect3{The <code>main</code> function}
+
+// The is the <code>main</code> function. We define here the polynomial degree
+// for the two finite element spaces (for the solution and the two liftings) and
+// the two penalty coefficients. We can also change the dimension to run the
+// code in 3D.
+int main()
+{
+  try
+    {
+      const unsigned int degree =
```

Comment:
Would it be an option to make n_refinements also a parameter here. So one only has to modify the parameters at one place to reproduce the tables in the result section.

Reply:
Sure, I made the update ;-).